docs/backend-system/building-plugins-and-modules/08-migrating.md

```diff
@@ -73,128 +73,102 @@ export const kubernetesPlugin = createBackendPlugin({
           config,
           logger,
           catalogApi,
           discovery,
         })
-          .setClusterSupplier(options.clusterSupplier)
+          .setClusterSupplier(configuredClusterSupplier)
           .build();
         http.use(router);
       },
     });
   },
-}));
-```
-
-The above would allow users to specify their own `ClusterSupplier` implementation to the plugin like this:
-
-```ts
-backend.add(
-  kubernetesPlugin({ clusterSupplier: new MyCustomClusterSupplier() }),
-);
+});
 ```
 
-Just to echo what was said above, this is not a very flexible solution and will for example be problematic to keep backwards compatible if we start evolving the options to for example accept multiple suppliers or tweak the `ClusterSupplier` interface.
-
-The new [extension points](../architecture/05-extension-points.md) API allows [modules](../architecture/06-modules.md) to add functionality into the backend plugin itself, in this case an additional `ClusterSupplier`.
-
-The kubernetes backend plugin only supports one `ClusterSupplier` at this time but let's look at how we could add support for multiple suppliers using extension points. This allows users to install several modules that add their own `ClusterSupplier` implementations to the plugin like this:
+There are however many types of customizations that are not possible to do with static configuration. In this case we want integrators to be able to create arbitrary implementations of the `ClusterSupplier` interface, which in the end requires an implementation through code. This is where the new backend system's [extension points](../architecture/05-extension-points.md) come in handy.
 
-```ts
-backend.add(kubernetesPlugin());
-backend.add(kubernetesGoogleContainerEngineClusterSupplier());
-backend.add(kubernetesElasticContainerEngine());
-```
-
-Now let's look at how to implement this with extension points. First we need to define the extension point itself. As the extension point will be used by other modules, it's common practice to export these from a shared package so that they can be imported by other modules and plugins.
+The new [extension points](../architecture/05-extension-points.md) API allows [modules](../architecture/06-modules.md) to add functionality into the backend plugin itself, in this case an additional `ClusterSupplier`. Let's look at how we could add support for installing custom suppliers using an extension point. This will allow integrators to build their own internal module with a custom `ClusterSupplier` implementation.
 
-We'll go ahead and create a `@backstage/plugin-kubernetes-node` package for this and from there we'll export the extension point.
+First we'll go ahead and create a `@backstage/plugin-kubernetes-node` package where we can defined our extension point. A separate package is used to avoid direct dependencies on the plugin package itself. With the new package created, we defined the extension point like this:
```

Comment:
```suggestion
First we'll go ahead and create a `@backstage/plugin-kubernetes-node` package where we can define our extension point. A separate package is used to avoid direct dependencies on the plugin package itself. With the new package created, we define the extension point like this:
```